src/main/java/com/aws/greengrass/util/platforms/windows/WindowsRunasProcess.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util.platforms.windows;
+
+import com.sun.jna.platform.win32.WinBase;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import lombok.Setter;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Manage a Windows process running as a specific user.
+ */
+@SuppressWarnings("PMD")  // TODO re-enable warnings
+@SuppressFBWarnings
+public class WindowsRunasProcess extends Process {
+    private final String domain;
+    private final String username;
+    private final String password;
+    @Setter
+    private String lpEnvironment;  // env vars in lpEnvironment format
+    @Setter
+    private String lpCurrentDirectory;
+
+    private WinBase.PROCESS_INFORMATION procInfo;
+    private boolean hasExited;
+    private int exitCode;
+
+    private InputStream inputStream;    // for reading child process stdout
+    private InputStream errorStream;    // for reading child process stderr
+    private OutputStream outputStream;  // for writing to child process stdin
+
+    /**
+     * Setup to run a Windows process as a specific user by providing credentials required by CreateProcessWithLogonW.
+     * @param domain name of domain that contains the user account
+     * @param username name of the account
+     * @param password clear-text password of the account
+     */
+    public WindowsRunasProcess(String domain, String username, String password) {
+        this.domain = domain;
+        this.username = username;
+        this.password = password;
+    }
+
+    public void start(String command) {
+        // TODO CreateProcessWithLogonW then in a different thread: WaitForSingleObject, GetExitCodeProcess
+        // reference UNIXProcess.initStreams
+    }
+
+    @Override
+    public OutputStream getOutputStream() {
+        return outputStream;
+    }
+
+    @Override
+    public InputStream getInputStream() {
+        return inputStream;
+    }
+
+    @Override
+    public InputStream getErrorStream() {
+        return errorStream;
+    }
+
+    @Override
+    public synchronized int waitFor() throws InterruptedException {
+        while (!hasExited) {
+            wait();
+        }
```

Comment:
not great... but this is all TODO I assume

Reply:
Draft for now... Copied this from java.lang.UNIXProcess though.